tests/test_oidc.py

```diff
@@ -91,3 +93,38 @@ def it_is_possible_to_get_iat(self, mock_auth, client, db_session):
         assert response.status_code == 302
 
         assert session['iat'] == expected_issue_timestamp
+
+
+def mock_requests_get(*args, **kwargs):
+    class MockResponse:
+
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    if 'https://login.microsoftonline.com/testid' in args[0]:
+        return MockResponse({
+            'authorization_endpoint':
+            'https://login.microsoftonline.com/testid/oauth2/authorize',
+            'discovery_url':
+            'https://login.microsoftonline.com/testid'}, 200)
+    else:
+        return MockResponse({}, 404)
+
+
+class WhenUserAuthenticatesWithAzureADProvider(object):
+
+    @patch('lib.oidc_old.requests.get', mock_requests_get)
+    def it_retrieves_correct_authorization_endpoint(self, app):
+        app.config["OIDC_PROVIDERS"] = {
+            'dex': {'discovery_url': 'http://dex.example.com:5556'},
+            'azure_ad': {'discovery_url': 'https://login.microsoftonline.'
+                         'com/testid'}
```

Comment:
Arbitrary indentation level. Break the line after the opening curly brace and indent one level.